blas/src/KokkosBlas2_ger.hpp

```diff
@@ -0,0 +1,146 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSBLAS2_GER_HPP_
+#define KOKKOSBLAS2_GER_HPP_
+
+#include <KokkosBlas2_ger_spec.hpp>
+
+namespace KokkosBlas {
+
+/// \brief Rank-1 update of a general matrix: A = A + alpha * x * y^{T,H}.
+///
+/// \tparam XViewType Input vector, as a 1-D Kokkos::View
+/// \tparam YViewType Input vector, as a 1-D Kokkos::View
+/// \tparam AViewType Input/Output matrix, as a 2-D Kokkos::View
+///
+/// \param space [in]     Execution space instance on which to run the kernel.
+///                       This may contain information about which stream to
+///                       run on.
+/// \param trans [in]     "T" or "t" for transpose, "H" or "h" for Hermitian.
+///                       Only the first character is taken into account.
+/// \param alpha [in]     Input coefficient of x * y^{T,H}
+/// \param x     [in]     Input vector, as a 1-D Kokkos::View
+/// \param y     [in]     Input vector, as a 1-D Kokkos::View
+/// \param A     [in/out] Output matrix, as a nonconst 2-D Kokkos::View
+template <class XViewType, class YViewType, class AViewType>
```

Comment:
Add a template parameter for the execution space: `class execution_space`, the user might want to specify it independently of what A's execution space is.

Reply:
Added. Thank you for the hint.